tests/maker-cli.rs

```diff
@@ -0,0 +1,335 @@
+
+        // Wait final setup
+        loop {
+            let log_message = maker_log_recvr.recv().unwrap();
+            if log_message.contains("Maker setup is ready") {
+                break;
+            }
+        }
+
+        log::info!("Maker setup is ready");
+
+        (maker_log_recvr, makerd_process, directoryd_process)
+    }
+
+    fn execute_maker_cli(&self, args: &[&str]) -> String {
+        let output = Command::new("./target/debug/maker-cli")
+            .args(args)
+            .output()
+            .unwrap();
+
+        // Capture the standard output and error from the command execution
+        let mut value = output.stdout;
+        let error = output.stderr;
+
+        // Panic if there is any error output
+        if !error.is_empty() {
+            panic!("Error: {:?}", String::from_utf8(error).unwrap());
+        }
+
+        // Remove the `\n` at the end of the output
+        value.pop();
+
+        // Convert the output bytes to a UTF-8 string
+        let output_string = std::str::from_utf8(&value).unwrap().to_string();
+
+        output_string
+    }
+}
+
+#[test]
+fn test_makecli_get_new_address() {
+    let maker_cli = MakerCli::new();
+    let (rx, mut makerd_proc, mut directoryd_proc) = maker_cli.start_makerd();
+
+    // Address check
+    let addr_resp = maker_cli.execute_maker_cli(&["new-address"]);
+    loop {
+        let log_message = rx.recv().unwrap();
+        if log_message.contains(" RPC request received: NewAddress") {
+            log::info!("RPC Message received");
+            break;
+        }
+    }
+    assert!(Address::from_str(&addr_resp).is_ok());
+
+    // Balance checks
+    let seed_balance = maker_cli.execute_maker_cli(&["seed-balance"]);
+    loop {
+        let log_message = rx.recv().unwrap();
+        if log_message.contains(" RPC request received: SeedBalance") {
+            log::info!("RPC Message received");
+            break;
+        }
+    }
+    let contract_balance = maker_cli.execute_maker_cli(&["contract-balance"]);
+    loop {
+        let log_message = rx.recv().unwrap();
+        if log_message.contains(" RPC request received: ContractBalance") {
+            log::info!("RPC Message received");
+            break;
+        }
+    }
+    let fidelity_balance = maker_cli.execute_maker_cli(&["fidelity-balance"]);
+    loop {
+        let log_message = rx.recv().unwrap();
+        if log_message.contains(" RPC request received: FidelityBalance") {
+            log::info!("RPC Message received");
+            break;
+        }
+    }
+    let swap_balance = maker_cli.execute_maker_cli(&["swap-balance"]);
+    loop {
+        let log_message = rx.recv().unwrap();
+        if log_message.contains(" RPC request received: SwapBalance") {
+            log::info!("RPC Message received");
+            break;
+        }
+    }
+    assert_eq!(seed_balance, "1000000 sats");
+    assert_eq!(swap_balance, "0 sats");
+    assert_eq!(fidelity_balance, "5000000 sats");
```

Comment:
Inside the `start_makerd` api -> we fund the maker wallet with `0.1 BTC` -> then we spent `0.05 BTC` for `Fidelity Bonds` + `1000 sats` for miner fees -> then how we can get the seed-balance equals to the amount we fund the wallet.
I guess, there is some syncing problem.

Reply:
yes.. that should not happen.. worth looking into it..